### PR TITLE
Zipcode 📍

### PR DESCRIPTION
**Pull request by SchweisguthN**

### CODE REVIEW

**Comment by SchweisguthN** [comment]
[ETQ U HubEE, je vois un nouveau formulaire pour la nouvelle démarche (CNAF Communes)](https://trello.com/c/L2iDufrf/1576-etq-u-hubee-je-vois-un-nouveau-formulaire-pour-la-nouvelle-d%C3%A9marche-cnaf-communes)

**Comment by Isalafont** [comment]
[Patch Monday !](https://trello.com/c/GtvZKyRv/1710-patch-monday)

**Comment by pierreericgarcia** [comment]
[ETQ instructeur du formulaire HubEE-CNAF, je vois le numéro de département de la commune demandeuse](https://trello.com/c/I12s5w6C/1670-etq-instructeur-du-formulaire-hubee-cnaf-je-vois-le-num%C3%A9ro-de-d%C3%A9partement-de-la-commune-demandeuse)

**Comment by Isalafont** [comment]
There's a new error message in the console on the Instructor Enrollement List: 

Capture d’écran 2023-01-25 à 11 03 36

File: backend/app/models/enrollment/hubee_portail_cnaf.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Enrollment::HubeePortailCnaf < Enrollment::HubeePortail
+  def submit_validation
+    errors.add(:zip_code, :invalid, message: "Vous devez renseigner un code postal valide avant de continuer") unless zip_code
+
+    if scopes.empty?
```
[the remark this answers is not in the document]
Comment (Isalafont) [reply backend/app/models/enrollment/hubee_portail_cnaf.rb:5]: My bad... 
we have to overwrite the `submit_validation` method.

So in this case, be careful, we can submit an enrollment without filling the `administrateur metier` and without selecting `CGU and DPO` case.

File: frontend/src/components/templates/InstructorEnrollmentList.tsx [hunk 2]
```diff
@@ -94,6 +95,45 @@ const InstructorEnrollmentList: React.FC = () => {
...
+    filterFn: 'includesString',
+    meta: {
+      placeholder: 'ex : 75009',
+    },
+  });
+
+  const changingColumn = user?.roles.some((role) =>
+    role.endsWith('hubee_portail_cnaf:reporter')
+  )
+    ? zipCodeColumn
```
Comment (Isalafont) [inline frontend/src/components/templates/InstructorEnrollmentList.tsx:134]: If we are administrator of all api, we have the column targetApiColumn who just disappears. It does not include much people: mostly : us (devs, Nico and Anais). 
We are sure that's the require behaviour that we want ?

File: frontend/src/config/data-provider-configurations.tsx [hunk 3]
```diff
@@ -122,6 +123,13 @@ export const DATA_PROVIDER_CONFIGURATIONS: {
     component: HubeePortailDila,
     url: 'https://portail.hubee.numerique.gouv.fr/',
   },
+  hubee_portail_cnaf: {
+    label: 'Portail HubEE - Démarches CNAF',
+    icon: 'logo-hubee.png',
+    email: 'x',
```
Comment (Isalafont) [inline frontend/src/config/data-provider-configurations.tsx:129]: should we not be an empty string here ?

**Review by Isalafont — COMMENTED** [review]
Nice PR ! 

I have a question, wasn't it possible to get the zipCode directly from the organizations of the Enrollment. Maybe in using an helper method directly from the `Enrollment` model and then pass it to the serializer so the frontend can access it. 

Might be a way to not having a new migration and not having a new section in the form ? (saving lot of code lines. ) 

What do you think ?

File: backend/app/models/enrollment.rb [hunk 4]
```diff
@@ -272,6 +272,7 @@ def self.csv_attributes
       demandeur_email
       team_members_json
       cgu_approved
+      zip_code
```
[the remark this answers is not in the document]
Comment (pierreericgarcia) [reply backend/app/models/enrollment.rb:275]: Good feedback !

File: frontend/src/components/templates/InstructorEnrollmentList.tsx (hunk 2 above)
Comment (pierreericgarcia) [reply frontend/src/components/templates/InstructorEnrollmentList.tsx:134]: As seen with @SchweisguthN, we can agree that yes it's the desired behaviour.

**Review by Isalafont — APPROVED** [review]

File: backend/app/services/filter_service.rb [hunk 5]
```diff
@@ -9,7 +9,7 @@ def is_valid_key(key, table_name)
     when "users"
       %w[email].include? key
     when "enrollments"
-      %w[id siret nom_raison_sociale target_api status team_members.email only_with_unprocessed_messages].include? key
+      %w[id siret nom_raison_sociale target_api status zip_code team_members.email only_with_unprocessed_messages].include? key
```
Comment (Isalafont) [inline backend/app/services/filter_service.rb:12]: zipcode filter is not working ? 

https://user-images.githubusercontent.com/43042737/223136098-82e6c945-b63d-4667-9249-f57aa9123199.mov

**Review by Isalafont — CHANGES_REQUESTED** [review]

File: backend/app/services/filter_service.rb (hunk 5 above)
Comment (Isalafont) [reply backend/app/services/filter_service.rb:12]: There is also a warning in the `user list` 

Is it the same for you ? 
Capture d’écran 2023-03-06 à 15 25 22

**Pull request merged**